### PR TITLE
Enable filename expansion on all completion contexts

### LINKED ISSUE #230: kubectl の kubeconfig オプションに `$HOME` や `~` を渡すと補完が動作しない
### OS version

```shell
Ubuntu 22.04.2 LTS (WSL 2)
```

### zsh version

```shell
zsh 5.8.1 (x86_64-ubuntu-linux-gnu)
```

### awk version

```shell
GNU Awk 5.1.0, API: 3.0 (GNU MPFR 4.1.0, GNU MP 6.2.1)
```

### jq version

```shell
jq-1.6
```

### fzf version

```shell
0.38.0 (352ea07)
```

### Command version(s)

```shell
git version 2.34.1
```

### Locale

```shell
LANG=C.UTF-8
LANGUAGE=
LC_CTYPE="C.UTF-8"
LC_NUMERIC="C.UTF-8"
LC_TIME="C.UTF-8"
LC_COLLATE="C.UTF-8"
LC_MONETARY="C.UTF-8"
LC_MESSAGES="C.UTF-8"
LC_PAPER="C.UTF-8"
LC_NAME="C.UTF-8"
LC_ADDRESS="C.UTF-8"
LC_TELEPHONE="C.UTF-8"
LC_MEASUREMENT="C.UTF-8"
LC_IDENTIFICATION="C.UTF-8"
LC_ALL=
```

### Aliases (if applicable)

```shell
~ ❯ alias                                                                                                         ✘ INT
k=kubectl
ktest='kubectl --kubeconfig /home/atis/.kube/config'
la='ls -lha'
ll='ls -lh'
ls='ls --color=auto'
which-command=whence
~ ❯ cat ~/.zshrc | grep alias
alias ll='ls -lh'
alias la='ls -lha'
alias k='kubectl'
alias ktest="kubectl --kubeconfig $HOME/.kube/config"
  alias ls='ls -G'
  alias ls='ls --color=auto'
```

### Installed zsh plugins (if applicable)

```shell
zsh-users/zsh-completions => as:"plugin"
docker/cli => as:"plugin", depth:"1", use:"contrib/completion/zsh"
romkatv/powerlevel10k => as:"theme", depth:"1", use:"powerlevel10k.zsh-theme"
chitoku-k/fzf-zsh-completions => as:"plugin"
```

### What happened?

`kubectl` コマンドにおいて、`--kubeconfig` オプションに `~/hoge/fuga`, `$HOME/hoge/fuga` を渡すと補完が動作しない。
alias に設定している場合も同様。

追記: alias の場合は挙動が異なる。[補足](https://github.com/chitoku-k/fzf-zsh-completions/issues/230#issuecomment-1588855797)

スクリーンショット 2023-06-13 173019
スクリーンショット 2023-06-13 173129

### What did you expect to happen?

補完が動作すること。
（`get pods **<TAB>` においては Pod 名が補完されること。）

### Steps to reproduce

`kubectl --kubeconfig $HOME/hoge/fuga get pods **<TAB>`
`kubectl --kubeconfig ~/hoge/fuga get pods **<TAB>`

### PR DESCRIPTION
**Pull request by chitoku-k**
Closes #230

### CODE REVIEW

**Comment by hexium310** [comment]
Other commands should also need parsing arguments, e.g. `git -C <path>`. How about splitting it into another file?

**Comment by chitoku-k** [comment]
@hexium310 That's a sharp insight but I'd like to keep this as is for now since kubectl is the only command for which this plugin currently supports this kind of completion. There could be such an opportunity for refactoring once the other commands to have the same feature!

**Comment by chitoku-k** [comment]
@hexium310 Finally, I made up my mind to change all other commands and changed the way they handle arguments.

File: src/core/parser.zsh [hunk 1]
```diff
@@ -6,18 +6,10 @@ _fzf_complete_get_command_pos() {
...
-    shift 1
-    local arguments=("${(Q)${(z)@}[@]}")
-    echo ${(q)arguments[$command_pos, -1]}
-}
-
-_fzf_complete_get_env() {
-    local command_pos=$1
-    shift 1
-    local arguments=("${${(z)@}[@]}")
-    echo ${arguments[1, $command_pos - 1]}
```
Comment (hexium310) [inline src/core/parser.zsh:20]: Do you need to remove those functions? I think that doing `[pos, -1]` or `[1, pos -1]` to trim/get env each time in the head of the complete functions is redundant and difficult to understand.

**Comment by hexium310** [comment]
I expect the function to expand a path in arguments.Does it make sense for the re-quoting function🤔

**Comment by chitoku-k** [comment]
> I expect the function to expand a path in arguments. Does it make sense for the function to re-quote🤔

According to **zshoptions(1)**, in order for the `MAGIC_EQUAL_SUBST` to work in parameter expansion, a substituted parameter has to be a unquoted shell argument that appears after the command name.

> **MAGIC_EQUAL_SUBST**
> All unquoted arguments of the form `anything=expression` appearing after the command name have filename expansion (that is, where *expression* has a leading `~` or `=`) performed on *expression* as if it were a parameter assignment.

```zsh
opt='--kubeconfig=~/.kube/config'
array=(${~opt})
echo ${array[@]} # => --kubeconfig=~/.kube/config
```

```zsh
opt='--kubeconfig=~/.kube/config'
echo ${~opt} # => --kubeconfig=/home/chitoku/.kube/config
```

That being said, taking into account as to how the `MAGIC_EQUAL_SUBST` option works, any command that returns its arguments is required along with the `GLOB_SUBST` option in parameter expansion (`~`).

So, why does it have to have such a function that re-quotes the arguments instead of just calling `echo`? -- The short answer is due to the word splitting. When the parameter expansion generates a word that contains a space, it gets split by spaces.

```zsh
opt='*'
touch 'foo bar' 'baz'

filenames=($(echo ${~opt}))
echo $#filenames # => 3
echo $filenames[1] # => baz
echo $filenames[2] # => foo
echo $filenames[3] # => bar

filenames=("$(echo ${~opt})")
echo $#filenames # => 1
echo $filenames[1] # => baz foo bar
```

Re-quoting the arguments solves this problem while it just requires that they be unquoted when calling it.

File: src/core/parser.zsh (hunk 1 above)
Comment (chitoku-k) [reply src/core/parser.zsh:20]: Actually, no. However, now that the logic is exactly the same when retrieving env part and command+argument part, it is not as difficult to understand as it was before, isn't it?

**Comment by hexium310** [comment]
I see, I take it that re-quoting is needed to remove needless quotes and only to escape the bare minimum.

**Review by hexium310 — APPROVED** [review]

**Pull request merged**